data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -152,26 +152,35 @@ private ParsedMessage convertS3EventMessages(final Message message) {
     }
 
     private List<DeleteMessageBatchRequestEntry> processS3EventNotificationRecords(final Collection<ParsedMessage> s3EventNotificationRecords) {
-        List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntryCollection = new ArrayList<>();
+        final List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntryCollection = new ArrayList<>();
+        final List<ParsedMessage> parsedMessagesToRead = new ArrayList<>();
+
         for (ParsedMessage parsedMessage : s3EventNotificationRecords) {
             if (parsedMessage.failedParsing) {
                 sqsMessagesFailedCounter.increment();
                 if (s3SourceConfig.getOnErrorOption().equals(OnErrorOption.DELETE_MESSAGES)) {
                     deleteMessageBatchRequestEntryCollection.add(buildDeleteMessageBatchRequestEntry(parsedMessage.message));
                 }
-            }
-            else {
+            } else {
                 final List<S3EventNotification.S3EventNotificationRecord> notificationRecords = parsedMessage.notificationRecords;
-                if(!notificationRecords.isEmpty() && isEventNameCreated(notificationRecords.get(0))) {
-                    final S3ObjectReference s3ObjectReference = populateS3Reference(notificationRecords.get(0));
-                    final Optional<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntry = processS3Object(parsedMessage, s3ObjectReference);
-                    deleteMessageBatchRequestEntry.ifPresent(deleteMessageBatchRequestEntryCollection::add);
+                if (!notificationRecords.isEmpty() && isEventNameCreated(notificationRecords.get(0))) {
+                    parsedMessagesToRead.add(parsedMessage);
                 } else {
                     // Add SQS message to delete collection if the eventName is not ObjectCreated
                     deleteMessageBatchRequestEntryCollection.add(buildDeleteMessageBatchRequestEntry(parsedMessage.message));
                 }
             }
         }
+
+        LOG.info("Received {} messages from SQS. Read {} messages from S3.", s3EventNotificationRecords.size(), parsedMessagesToRead.size());
```

Comment:
Would we be better emitting a metric over logging?

Reply:
I think we do emit metrics here. I've found this logging helpful in some of the work I've been trying to do. Data Prepper administrators can always increase the log level. I'm also interested in the possibility of logging the S3 objects (as there are no metrics for that).